core/src/test/resources/sql/sub-query.iq

```diff
@@ -830,7 +830,7 @@ EnumerableCalc(expr#0..3=[{inputs}], expr#4=[IS NULL($t3)], expr#5=[false], expr
     EnumerableLimit(fetch=[1])
       EnumerableSort(sort0=[$0], dir0=[DESC])
         EnumerableAggregate(group=[{0}], c=[COUNT()])
-          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[false], expr#4=[null], expr#5=[IS NULL($t4)], cs=[$t3], $condition=[$t5])
+          EnumerableCalc(expr#0..2=[{inputs}], expr#3=[false], expr#4=[123], expr#5=[null], expr#6=[=($t4, $t5)], expr#7=[IS NULL($t5)], expr#8=[OR($t6, $t7)], cs=[$t3], $condition=[$t8])
```

Comment:
This change seem odd.
Why `IS NULL($t5)` was not simplified to false in this case?
Why `=($t4, $t5)` was not simplified to null?

Reply:
I think I know where it comes from. The rule FilterProjectTransposeRule creates a newCondition (line 115) that in some cases (line 121) it is not simplified. Since we cannot really control the order that the rules are firing we may end up with cases like the above.

Reply:
Technically speaking, there should be `ReduceExpressionsRule` which should simplify all the things, however it is not enabled by default for some reason: https://github.com/apache/calcite/commit/221739354b56e34e9f1d41b42a0e6881a8f5ddee#diff-6a278acbc7f012ba62bdc1251b70ba25R179

However, it looks like it makes sense to add relevant simplification to FilterProjectTransposeRule